india_compliance/gst_india/overrides/payment_entry.py

```diff
@@ -14,6 +16,40 @@
 from india_compliance.gst_india.utils import get_all_gst_accounts
 
 
+@frappe.whitelist()
+def get_reconciliation_status_for_invoice_list(invoice_list):
+    if not invoice_list:
+        return
+
+    if isinstance(invoice_list, str):
+        invoice_list = json.loads(invoice_list)
+
+    original_list = frappe.db.get_list(
+        "Purchase Invoice",
+        filters={"name": ["in", invoice_list]},
+        fields=("name", "reconciliation_status"),
+    )
+    transformed_dict = {
+        item["name"]: item["reconciliation_status"] for item in original_list
+    }
```

Comment:
You don't need dict comprehension. This should do the trick:

`dict(frappe.get_list(..., ..., as_list=True))`